File: TestChapChap.java
import java.net.URL;
import java.net.HttpURLConnection;
import java.io.BufferedReader;
import java.io.InputStreamReader;
public class TestChapChap {
    public static void main(String[] args) {
        try {
            URL url = new URL("https://chapchappay.com/api/ecommerce/order/SUB_123456789");
            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
            connection.setRequestMethod("GET");
            System.out.println(connection.getResponseCode());
            BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream()));
            String inputLine;
            while ((inputLine = in.readLine()) != null) {
                System.out.println(inputLine);
            }
            in.close();
        } catch(Exception e) {
            e.printStackTrace();
        }
    }
}

File: test_chapchap.kt
import java.net.URL
import java.net.HttpURLConnection
fun main() {
    val url = URL("https://chapchappay.com/api/ecommerce/order/SUB_123456789")
    val connection = url.openConnection() as HttpURLConnection
    connection.requestMethod = "GET"
    // connection.setRequestProperty("CCP-Api-Key", "dummy")
    try {
        println(connection.responseCode)
        println(connection.inputStream.bufferedReader().use { it.readText() })
    } catch(e: Exception) {
        println(connection.errorStream?.bufferedReader()?.use { it.readText() })
    }
}
